pymc/backends/zarr.py

```diff
@@ -0,0 +1,432 @@
+        draw = len(self.root[f"full_{group_name}"].draw)
+        if tune > 0:
+            store_warmup = True
+            warmup_group = self.root.create_group(f"{WARMUP_TAG}{group_name}", overwrite=True)
+        if tune < draw:
+            store_posterior = True
+            posterior_group = self.root.create_group(f"{group_name}", overwrite=True)
+        for name, array in self.root[f"full_{group_name}"].arrays():
+            if name == "draw":
+                if store_warmup:
+                    warmup_array = warmup_group.array(
+                        name="draw", data=np.arange(tune), dtype="int"
+                    )
+                if store_posterior:
+                    posterior_array = posterior_group.array(
+                        name="draw", data=np.arange(len(array) - tune), dtype="int"
+                    )
+            else:
+                dims = array.attrs["_ARRAY_DIMENSIONS"]
+                if len(dims) >= 2 and dims[:2] == ["chain", "draw"]:
+                    warmup_idx = (slice(None), slice(None, tune, None))
+                    posterior_idx = (slice(None), slice(tune, None, None))
+                else:
+                    warmup_idx = slice(None)
+                    posterior_idx = slice(None)
+                fill_value, dtype, object_codec = get_initial_fill_value_and_codec(array.dtype)
+                if store_warmup:
+                    warmup_array = warmup_group.array(
+                        name=name,
+                        data=array[warmup_idx],
+                        dtype=dtype,
+                        fill_value=fill_value,
+                        object_codec=object_codec,
+                    )
+                if store_posterior:
+                    posterior_array = posterior_group.array(
+                        name=name,
+                        data=array[posterior_idx],
+                        dtype=dtype,
+                        fill_value=fill_value,
+                        object_codec=object_codec,
+                    )
+            if store_warmup:
+                warmup_array.attrs.update(array.attrs.asdict())
+            if store_posterior:
+                posterior_array.attrs.update(array.attrs.asdict())
+
+    def to_inferencedata(self) -> az.InferenceData:
+        self.consolidate()
+        # The ConsolidatedMetadataStore looks like an empty store from xarray's point of view
+        # we need to actually grab the underlying store so that xarray doesn't produce completely
+        # empty arrays
+        store = self.root.store.store
+        groups = {}
+        try:
+            global_attrs = {
+                "tuning_steps": self.tuning_steps,
+                "sampling_time": self.sampling_time,
+            }
+        except AttributeError:
+            global_attrs = {}  # pragma: no cover
+        for name, _ in self.root.groups():
+            if name not in SUPPORTED_GROUPS_ALL:
+                continue
```

Comment:
I would instead exclude whatever groups are considered special/private by pymc. The supported groups concept will be removed as it goes against the general idea with inferencedata (if things with a specific name are available they will be used and interpreted in the way defined in the schema, but storing arbitrary things that help you in there is also perfectly valid).

The warning was useful at first when actively developing converters and for developers to easily catch having named a group `poterior` instead of `posterior` but now that inferencedata uses are more advanced, the warning is very annoying.

Reply:
Thanks for letting me know. I'll go back to my original approach which was to only store groups that didn't start with an underscore.